chain/client/src/test_utils/test_env.rs

```diff
@@ -312,66 +319,42 @@ impl TestEnv {
                             &state_witness,
                         );
 
-                    output.chunk_hash_to_account_ids.insert(
-                        state_witness.inner.chunk_header.chunk_hash(),
-                        account_ids.into_iter().collect(),
-                    );
                     None
                 }
                 _ => Some(request),
             });
         }
+
+        // Wait for all state witnesses to be processed before returning.
+        for processing_done_waiter in witness_processing_done_waiters {
```

Comment:
Shouldn't we return this and let the caller handle the waiting? Technically this function is just to propagate state witness and not to wait for completion. We can let the `propagate_chunk_endorsements` do the waiting?

I'm fine either way, leaving the final implementation to you!

Reply:
It does simplify `propagate_chunk_endorsements` most definitely, so the current implementation may also be good!

Reply:
For the current use cases it's more convenient to have a function that waits for them to be processed before returning, so that's what I went with. If we ever need to do something else, we could write another function for that.

I think it's pretty consistent with the rest of the code, `propagate_chunk_endorsements` also "waits" for the endorsements to be processed, functions that don't wait for results usually start with `start_`, like `start_process_block`.